rotary/rotary.py

```diff
@@ -136,6 +139,41 @@ def init(args):
     dataset.create_sample_tsv(output_dir_path, header=sample_tsv_header_fields)
 
 
+def download(args):
+    """
+    Runs code for the download command-line mode. Downloads database files required by rotary.
+
+    :param args: The command-line arguments.
+    """
+    config_path = get_config_path(args, default_config_path=provided_config_path)
```

Comment:
@LeeBergstrand When I reviewed the code, I was a little confused about how the config file is handled. It looks like this function (and the rotary CLI) can accept a user's config file as input. However, this function also writes `download_config.yaml` as output, and it seems like it can generate `download_config.yaml` using the rotary default config if the user doesn't supply a config file. Would it be better to not write `download_config.yaml` if a user-supplied config is input? Or (as an alternative approach) would it be better to just eliminate the option to provide a user-supplied config file? I'm not sure what info the user could  supply in their config file that isn't already known by the download function. Let me know your thoughts or if I am misunderstanding how the code works -- thanks.

Reply:
I do this to get the download dir path into the Snakemake workflow (as well as several other parameters like hmm URLs, etc.). You can potentially pass the environment variables in as CLI parameters to make Snakemake work without a config file, but this would require a second way of running Snakemake in Pungi.

Reply:
Thanks for the additional explanation. That makes sense that a config file needs to be written to disk to run the snakemake workflow. The thing I was wondering is that the CLI for the download module seems to have a flag where the user can provide their own config file. What happens in that case? Does it override the temp config file, for example? Or am I misunderstanding the code somehow? I'll start a 2nd thread where I highlight that code line (252-ish) specifically. Thanks!

Reply:
Addressed